vision_agent/utils/execute.py

```diff
@@ -309,6 +309,21 @@ def success(self) -> bool:
         """
         return self.error is None
 
+    def get_main_result(self) -> Optional[Result]:
+        """
+        Get the main result of the execution.
+        An execution may have multiple results, e.g. intermediate outputs. The main result is the last output of the cell execution.
+        """
+        if not self.success:
+            _LOGGER.info("Result is not available as the execution was not successful.")
+            return None
+        if not self.results or not any(res.is_main_result for res in self.results):
+            _LOGGER.info("Execution was successful but there is no main result.")
+            return None
+        main_result = self.results[-1]
+        assert main_result.is_main_result, "The last result should be the main result."
```

Comment:
Do you want this as an assert? or just have it raise an exception or something, seems like it's still fine if it runs execute but doesn't return the result (shouldn't crash)